helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -505,6 +505,46 @@ public void setMaxConcurrentTask(int maxConcurrentTask) {
     _record.setIntField(InstanceConfigProperty.MAX_CONCURRENT_TASK.name(), maxConcurrentTask);
   }
 
+  /**
+   * Get the instance capacity information from the map fields
+   * @return data map if it exists, or empty map
+   */
+  public Map<String, Integer> getInstanceCapacityMap() {
+    Map<String, String> capacityData = _record.getMapField(InstanceConfigProperty.INSTANCE_CAPACITY_MAP.name());
```

Comment:
Can you enhance the checks in _record.getMapFields? E.g if we know sure the worst case of returned map is empty map which cannot be null. A lot of redundant checks here and there can be avoided.

Reply:
The check logic depends on business logic. Like in REST service all the API check the input themselves. Not quite possible to abstract the check-in getMap.

Unless we make multiple getMap calls. Might be an option. Let me consider after this PR.

Reply:
Current getMapField impl is  
```
public Map<String, String> getMapField(String k) {
    return mapFields.get(k);
  }
```
if you make it 
```
public Map<String, String> getMapField(String k) {
    return mapFields.getOrDefault(k, new HashMap<>());
  }
```
I don't think it cause backward compability issue and you won't need to check if map == null everywhere. For those who already checks, it doesn't hurt.

Reply:
This change requires us to revisit all the callers. Some of them assume a null return. Let's do it separately. Not in this PR.